src/components/Carousel/Carousel.js

```diff
@@ -33,7 +34,10 @@ class Carousel extends Component {
     const { children: prevChildren } = this.props;
     const { autoPlayActive: prevAutoPlayActive } = this.state;
 
-    if (nextChildren !== prevChildren) {
+    const childrenChanged = nextChildren.length !== prevChildren.length ||
+      !isEqual(nextChildren.map(({ props }) => props), prevChildren.map(({ props }) => props));
```

Comment:
What do you think about working with `Array.every()` and `Object.values()` here? It could potentially improve the legibility of this line